### PR TITLE
[7671] replace django-background-tasks with celery

### PR DESCRIPTION
**Pull request by goapunk**

### CODE REVIEW

**Review by m4ra — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -8,12 +8,13 @@ based on [adhocracy 4](https://github.com/liqd/adhocracy4).
...
-*   libjpeg
-*   libpq (only if postgres should be used)
-*   gdal
+- nodejs (+ npm)
+- python 3.x (+ venv + pip)
+- libmagic
+- libjpeg
+- libpq (only if postgres should be used)
+- gdal
+- redis (only if celery is used)
```
[the remark this answers is not in the document]
Comment (goapunk) [reply README.md]: changed, let me know if it's more clear now :)

File: Makefile [hunk 2]
```diff
@@ -62,9 +62,13 @@ help:
...
-	@echo "  make start-postgres			-- start the local postgres cluster"
-	@echo "  make stop-postgres				-- stops the local postgres cluster"
-	@echo "  make create-postgres			-- create the local postgres cluster (only works on ubuntu 20.04)"
+	@echo "  make postgres-start			-- start the local postgres cluster"
+	@echo "  make postgres-stop				-- stops the local postgres cluster"
+	@echo "  make postgres-create			-- create the local postgres cluster (only works on ubuntu 20.04)"
+	@echo "  make local-a4					-- patch to use local a4 (needs to have path ../adhocracy4)"
+	@echo "  make celery-worker-start		-- starts the celery worker in the foreground
+	@echo "  make celery-worker-status		-- lists all registered tasks and active worker nodes
+	@echo "  make celery-worker-dummy-task	-- calls the dummy task and prints result from redis
```
[the remark this answers is not in the document]
Comment (goapunk) [reply Makefile]: good catch

File: README.md [hunk 3]
```diff
@@ -24,13 +25,42 @@ based on [adhocracy 4](https://github.com/liqd/adhocracy4).
     make watch
 
 ### Use postgresql database for testing:
+
 run the following command once:
+
 ```
 make create-postgres
```
[the remark this answers is not in the document]
Comment (goapunk) [reply README.md]: fixed

File: docs/celery.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+Currently, we make use of the following config parameters:
+- `broker_url = "redis://localhost:6379"`
+- `result_backend = "redis"`
+- `broker_connection_retry_on_startup = True`
+- `result_extended = True`
+
+The celery app is configured from the django settings file and namespaced variables. The defaults are defined in `config/settings/base.py` but can be overriden by `config/settings/local.py`.
+
+Note that the default settings in`dev.py` enable the "always eager mode" (`CELERY_TASK_ALWAYS_EAGER = True`) in which the django server will run the shared tasks itself. This is to keep the development setup as lightweight as possible. If you want to develop or test using a celery worker make sure that you add the following to your `local.py`:
```
Comment (m4ra) [inline docs/celery.md:21]: can we add here --> and install and run server redis in your system, or something like this.

File: requirements/base.txt [hunk 5]
```diff
@@ -11,6 +11,8 @@ urllib3==2.0.3
 wagtail==4.1.6
 whitenoise==6.5.0
 zeep==4.2.1
+celery==5.3.1
```
Comment (m4ra) [inline requirements/base.txt:14]: celery is inherited from a4

File: docs/celery.md (hunk 4 above)
Comment (goapunk) [reply docs/celery.md:21]: sure, but this is just copied from the a+ PR, so we should probably change it there as well then

Comment (m4ra) [reply docs/celery.md:21]: yes, and the makefile quotes, and the readme, and the requirements base (redis goes our of a4 inherited packages)

**Review by m4ra — APPROVED** [review]
okay!

**Pull request merged**